vmware-event-router/internal/processor/aws_event_bridge.go

```diff
@@ -173,36 +173,38 @@ func NewAWSEventBridgeProcessor(ctx context.Context, cfg connection.Config, sour
 // throttling/batching
 // https://docs.aws.amazon.com/eventbridge/latest/userguide/cloudwatch-limits-eventbridge.html#putevents-limits
 func (awsEventBridge *awsEventBridgeProcessor) Process(moref types.ManagedObjectReference, baseEvent []types.BaseEvent) error {
-	input, err := awsEventBridge.createPutEventsInput(baseEvent)
+	batchInput, err := awsEventBridge.createPutEventsInput(baseEvent)
 	if err != nil {
 		awsEventBridge.Printf("could not create PutEventsInput for event(s): %v", err)
 		return nil
 	}
 
 	// nothing to send
-	if len(input.Entries) == 0 {
+	if len(batchInput) == 0 {
 		return nil
 	}
 
-	// TODO: investigate limits on number/size of entries in a single put
-	resp, err := awsEventBridge.PutEvents(&input)
-	if err != nil {
-		awsEventBridge.Printf("could not send event(s): %v", err)
-		return nil
-	}
-	if awsEventBridge.verbose {
-		awsEventBridge.Printf("successfully sent event(s) from source %s: %+v", awsEventBridge.source, resp)
+	for _, input := range batchInput {
+		resp, err := awsEventBridge.PutEvents(&input)
+		if err != nil {
+			awsEventBridge.Printf("could not send event(s): %v", err)
```

Comment:
Are we deciding on the first failure stops the whole iteration (batches)?

Reply:
I miscalculated that you are right we should not return, instead I have changed with `continue` :+1: